### ISSUE TITLE
Animated water looks absurdly strange from different angles

### ISSUE THREAD

**Issue by jeffswt** [MovingBlocks/Terasology#2450]
### What you were trying to do

Trying to turn on animated water and look nice.
### What actually happened

When I attempted to turn on the animated water (though successfully turned on), the behaviors of the animated water seemed strange. When looked from different viewports (e.g. from the north and the south), the level of the water varies differently, and constantly changes according to viewport. Probably because the water is rendered separately from the blocks and entities, it seems to have a de-sync between water rendering and block rendering, where they should stick to each other at borders.

In other words, at the borders where water meet blocks, from some angles of view we may see through the water (a gap between the water surface and the edge of the land).
### How to reproduce
- Turn on animated water
- View from different angles
### Log details and game version

Pre-Alpha #63 / Alpha 3.
### Computer details
- Intel Core 2 Duo T7500
- NVidia Quadro FX570M
- 1GB Java Heap Space

**A screenshot would be added later.**
##

**Comment by Cervator** [comment]
Hey there - yep, the animated water is meant to have "waves" which pretty much work, but do not connect well to the existing block grid on a beach, for instance. Same issue as #440 with trees and one suggestion was #504 to hide the gap using a crashing foam type effect when the waves hit land. There are probably other options as well.

Both those issues are sadly pretty old as we lost @begla who was the wizard of such things :-) Right now we're refactoring the rendering system to make it easier to implement new effects, actually have a Google Summer of Code student working on it. I'm still not sure when it'll be worked on though. But we may not have an explicit issue covering the request so this issue should work well for that, thanks!
